src/BaseElement.js

```diff
@@ -302,40 +305,40 @@ class BaseElement extends HTMLElement {
 		}
 	}
 
-	/**
-	 * Context Properties to issue Context Requests and to pull contextual properties into the elements scope
-	 * @return {{}}
-	 */
-	context() {
-		return {};
-	}
-
 	/**
 	 * Defines context on the element based on keys from this.context()
 	 */
 	defineContext() {
-		Object.entries(this.context()).forEach(([key, value]) => {
-			this.dispatch('request-context', { [key]: value }, true);
-		});
 		// define context provider
-		if (this._options.provideContext) {
+		if (Object.values(this._options.propertyOptions).find((option) => option.provideContext === true)) {
+			// at least one property that provides Context
 			this.addEventListener('request-context', this.onRequestContext);
 		}
 	}
 
+	doRequestContext(property, callback) {
+		this.dispatch('request-context', { [property]: callback }, true);
+	}
+
+	/**
+	 * catch context requests and provide data if there is some
+	 * */
 	onRequestContext(event) {
```

Comment:
@quarkus If we don't need to remove it, we could make it inline and don't need this odd name...

Reply:
For now i´d like to treat it just like the other events. Maybe sth. wo consider with v1 ?